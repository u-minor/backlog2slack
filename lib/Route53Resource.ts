import { Stack } from 'aws-cdk-lib';
import {
  AaaaRecord,
  ARecord,
  IHostedZone,
  RecordTarget,
} from 'aws-cdk-lib/aws-route53';

export interface Route53ResourceProps {
  hostedZone: IHostedZone;
  recordName: string;
  target: RecordTarget;
}

export class Route53Resource {
  constructor(stack: Stack, props: Route53ResourceProps) {
    const route53RecordProps = {
      zone: props.hostedZone,
      recordName: props.recordName,
      target: props.target,
    };

    new ARecord(stack, 'ARecord', route53RecordProps);
    new AaaaRecord(stack, 'AaaaRecord', route53RecordProps);
  }
}
