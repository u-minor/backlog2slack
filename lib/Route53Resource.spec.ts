import { App, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { HostedZone, RecordTarget } from 'aws-cdk-lib/aws-route53';
import { Route53Resource } from './Route53Resource';

describe('Route53Resource', () => {
  let template: Template;

  beforeAll(() => {
    const app = new App();
    const stack = new Stack(app, 'TestStack');
    const recordTarget = {
      aliasTarget: {
        bind: () => ({
          hostedZoneId: 'CLOUDFRONT_ZONE_ID',
          dnsName: 'foobarbaz.cloudfront.net.',
        }),
      },
    } as RecordTarget;

    new Route53Resource(stack, {
      hostedZone: {
        hostedZoneId: 'MY_ZONE_ID',
        zoneName: 'example.com',
      } as unknown as HostedZone,
      recordName: 'test.example.com',
      target: recordTarget,
    });

    template = Template.fromStack(stack);
  });

  it('should generate valid Route53 record set resources', () => {
    const actual = template.findResources('AWS::Route53::RecordSet');

    expect(actual).toMatchSnapshot();
  });
});
