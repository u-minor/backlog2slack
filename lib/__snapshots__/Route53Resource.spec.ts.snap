// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route53Resource should generate valid Route53 record set resources 1`] = `
{
  "ARecordE7B57761": {
    "Properties": {
      "AliasTarget": {
        "DNSName": "foobarbaz.cloudfront.net.",
        "HostedZoneId": "CLOUDFRONT_ZONE_ID",
      },
      "HostedZoneId": "MY_ZONE_ID",
      "Name": "test.example.com.",
      "Type": "A",
    },
    "Type": "AWS::Route53::RecordSet",
  },
  "AaaaRecord61D2056E": {
    "Properties": {
      "AliasTarget": {
        "DNSName": "foobarbaz.cloudfront.net.",
        "HostedZoneId": "CLOUDFRONT_ZONE_ID",
      },
      "HostedZoneId": "MY_ZONE_ID",
      "Name": "test.example.com.",
      "Type": "AAAA",
    },
    "Type": "AWS::Route53::RecordSet",
  },
}
`;
